```
Error running mint automation
+ mkdir -p 10623-c0bc7b2/gopath/src/github.com/minio
+ git -C 10623-c0bc7b2/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 10623-c0bc7b2/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 10623-c0bc7b2/gopath/src/github.com/minio/minio fetch --quiet minio pull/10623/head:pr10623
+ git -C 10623-c0bc7b2/gopath/src/github.com/minio/minio checkout --quiet pr10623
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/10623-c0bc7b2/gopath
+ make -C 10623-c0bc7b2/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/cli v1.22.0
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/minio/minio-go/v7 v7.0.5-0.20200811211821-14ed05478889
go: downloading github.com/colinmarc/hdfs/v2 v2.1.1
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/gorilla/handlers v1.4.2
go: downloading github.com/prometheus/client_golang v1.0.0
go: downloading github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading golang.org/x/net v0.0.0-20200904194848-62affa334b73
go: downloading go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: downloading github.com/cheggaaa/pb v1.0.28
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/mattn/go-runewidth v0.0.2
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading google.golang.org/grpc v1.26.0
go: downloading github.com/dchest/siphash v1.2.1
go: downloading google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: downloading github.com/klauspost/compress v1.10.3
go: downloading github.com/minio/sio v0.2.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/minio/simdjson-go v0.1.5
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/gogo/protobuf v1.3.1
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/klauspost/pgzip v1.2.1
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading golang.org/x/sys v0.0.0-20200915084602-288bc346aa39
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: downloading go.opencensus.io v0.22.3
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/minio/md5-simd v1.1.0
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/coreos/go-semver v0.2.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/willf/bitset v1.1.11
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/miekg/dns v1.1.8
go: downloading github.com/prometheus/common v0.4.1
go: downloading github.com/prometheus/procfs v0.0.8
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/olivere/elastic.v5 v5.0.86
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/mattn/go-isatty v0.0.8
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/Shopify/sarama v1.24.1
go: downloading github.com/beevik/ntp v0.2.0
go: downloading github.com/google/uuid v1.1.1
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/mattn/go-colorable v0.1.4
go: downloading github.com/coreos/go-systemd/v22 v22.0.0
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/klauspost/reedsolomon v1.9.9
go: downloading github.com/golang/protobuf v1.4.0
go: downloading go.uber.org/multierr v1.5.0
go: downloading github.com/lib/pq v1.8.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/beorn7/perks v1.0.0
go: downloading github.com/mailru/easyjson v0.7.1
go: downloading github.com/nats-io/nkeys v0.1.4
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/eapache/queue v1.1.0
go: downloading google.golang.org/protobuf v1.22.0
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading go.uber.org/atomic v1.6.0
go: downloading github.com/nats-io/stan.go v0.7.0
go: downloading github.com/golang/groupcache v0.0.0-20190702054246-869f871628b6
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/hashicorp/vault/api v1.0.4
go: downloading github.com/nats-io/jwt v0.3.2
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/eapache/go-resiliency v1.1.0
go: downloading github.com/tidwall/gjson v1.3.5
go: downloading github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/tinylib/msgp v1.1.2
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading github.com/tidwall/pretty v1.0.0
go: downloading github.com/tidwall/match v1.0.1
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/philhofer/fwd v1.0.0
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
# github.com/minio/minio/cmd
cmd/web-handlers.go:718:87: undefined: w
make: *** [build] Error 2

```

--- a/cmd/handler-utils.go
+++ b/cmd/handler-utils.go
@@ -238,7 +238,9 @@ func extractReqParams(r *http.Request) map[string]string {
 
 // Extract response elements to be sent with event notifiation.
 func extractRespElements(w http.ResponseWriter) map[string]string {
-
+	if w == nil {
+		return map[string]string{}
+	}
 	return map[string]string{
 		"requestId":      w.Header().Get(xhttp.AmzRequestID),
 		"content-length": w.Header().Get(xhttp.ContentLength),

--- a/cmd/object-handlers-common.go
+++ b/cmd/object-handlers-common.go
@@ -281,7 +281,7 @@ func setPutObjHeaders(w http.ResponseWriter, objInfo ObjectInfo, delete bool) {
 // deleteObject is a convenient wrapper to delete an object, this
 // is a common function to be called from object handlers and
 // web handlers.
-func deleteObject(ctx context.Context, obj ObjectLayer, cache CacheObjectLayer, bucket, object string, r *http.Request, opts ObjectOptions) (objInfo ObjectInfo, err error) {
+func deleteObject(ctx context.Context, obj ObjectLayer, cache CacheObjectLayer, bucket, object string, w http.ResponseWriter, r *http.Request, opts ObjectOptions) (objInfo ObjectInfo, err error) {
 	deleteObject := obj.DeleteObject
 	if cache != nil {
 		deleteObject = cache.DeleteObject
@@ -293,22 +293,24 @@ func deleteObject(ctx context.Context, obj ObjectLayer, cache CacheObjectLayer, 
 		if objInfo.DeleteMarker {
 			// Notify object deleted marker event.
 			sendEvent(eventArgs{
-				EventName:  event.ObjectRemovedDeleteMarkerCreated,
-				BucketName: bucket,
-				Object:     objInfo,
-				ReqParams:  extractReqParams(r),
-				UserAgent:  r.UserAgent(),
-				Host:       handlers.GetSourceIP(r),
+				EventName:    event.ObjectRemovedDeleteMarkerCreated,
+				BucketName:   bucket,
+				Object:       objInfo,
+				ReqParams:    extractReqParams(r),
+				RespElements: extractRespElements(w),
+				UserAgent:    r.UserAgent(),
+				Host:         handlers.GetSourceIP(r),
 			})
 		} else {
 			// Notify object deleted event.
 			sendEvent(eventArgs{
-				EventName:  event.ObjectRemovedDelete,
-				BucketName: bucket,
-				Object:     objInfo,
-				ReqParams:  extractReqParams(r),
-				UserAgent:  r.UserAgent(),
-				Host:       handlers.GetSourceIP(r),
+				EventName:    event.ObjectRemovedDelete,
+				BucketName:   bucket,
+				Object:       objInfo,
+				ReqParams:    extractReqParams(r),
+				RespElements: extractRespElements(w),
+				UserAgent:    r.UserAgent(),
+				Host:         handlers.GetSourceIP(r),
 			})
 		}
 	}

--- a/cmd/object-handlers.go
+++ b/cmd/object-handlers.go
@@ -2726,7 +2726,7 @@ func (api objectAPIHandlers) DeleteObjectHandler(w http.ResponseWriter, r *http.
 	}
 
 	// http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectDELETE.html
-	objInfo, err := deleteObject(ctx, objectAPI, api.CacheAPI(), bucket, object, r, opts)
+	objInfo, err := deleteObject(ctx, objectAPI, api.CacheAPI(), bucket, object, w, r, opts)
 	if err != nil {
 		switch err.(type) {
 		case BucketNotFound:

--- a/cmd/web-handlers.go
+++ b/cmd/web-handlers.go
@@ -715,7 +715,7 @@ next:
 				}
 			}
 
-			_, err = deleteObject(ctx, objectAPI, web.CacheAPI(), args.BucketName, objectName, r, opts)
+			_, err = deleteObject(ctx, objectAPI, web.CacheAPI(), args.BucketName, objectName, nil, r, opts)
 			logger.LogIf(ctx, err)
 			continue
 		}
@@ -771,12 +771,34 @@ next:
 			}
 
 			// Deletes a list of objects.
-			_, errs := deleteObjects(ctx, args.BucketName, objects, opts)
+			deletedObjects, errs := deleteObjects(ctx, args.BucketName, objects, opts)
 			for _, err := range errs {
 				if err != nil {
 					logger.LogIf(ctx, err)
 					break next
 				}
+			}
+			// Notify deleted event for objects.
+			for _, dobj := range deletedObjects {
+				objInfo := ObjectInfo{
+					Name:      dobj.ObjectName,
+					VersionID: dobj.VersionID,
+				}
+				if dobj.DeleteMarker {
+					objInfo = ObjectInfo{
+						Name:         dobj.ObjectName,
+						DeleteMarker: dobj.DeleteMarker,
+						VersionID:    dobj.DeleteMarkerVersionID,
+					}
+				}
+				sendEvent(eventArgs{
+					EventName:  event.ObjectRemovedDelete,
+					BucketName: args.BucketName,
+					Object:     objInfo,
+					ReqParams:  extractReqParams(r),
+					UserAgent:  r.UserAgent(),
+					Host:       handlers.GetSourceIP(r),
+				})
 			}
 		}
 	}

